ops.go

```diff
@@ -53,31 +53,31 @@ func Mv(r *Root, src, dst string) error {
 		return err
 	}
 
-	fsn, err := dstDir.Child(filename)
+	fsn, err := dstDir.Child(dstFname)
 	if err == nil {
 		switch n := fsn.(type) {
 		case *File:
-			_ = dstDir.Unlink(filename)
+			_ = dstDir.Unlink(dstFname)
 		case *Directory:
 			dstDir = n
-			filename = srcFname
+			dstFname = srcFname
 		default:
 			return fmt.Errorf("unexpected type at path: %s", dst)
 		}
 	} else if err != os.ErrNotExist {
 		return err
 	}
 
-	err = dstDir.AddChild(filename, nd)
+	err = dstDir.AddChild(dstFname, nd)
 	if err != nil {
 		return err
 	}
 
-	if srcDir == dstDirStr && srcFname == filename {
+	if srcDir.name == dstDir.name && srcFname == dstFname {
```

Comment:
This is the only substantive change in this file and is the actual source of the bug revealed in go-ipfs sharness tests. Everything else is naming changes (which I can totally revert if it's too noisy).